src/fsharp/lex.fsl

```diff
@@ -124,7 +124,7 @@ let startString args (lexbuf: UnicodeLexing.Lexbuf) =
                             BYTEARRAY (Lexhelp.stringBufferAsBytes buf)
                         )
                      else
-                        STRING (System.Text.Encoding.Unicode.GetString(s,0,s.Length)))  
+                        STRING (Lexhelp.stringBufferAsString s))  
```

Comment:
In initial testing, I computed the string both ways here and compared, to make sure there were no unexpected inconsistencies.

After running all of the tests, the only differences were cases where the old method was doing the wrong thing.  So regression risk should be low.